resources/analysisTools/qcPipeline/qcJson.pl

```diff
@@ -329,20 +329,20 @@ ($$)
 
 
 sub ensureInt ($) {
-    my ($val) = @_;
-    if ($val !~ /^\d$/) {
+	my ($val) = @_;
+	if ($val !~ /^(?:\d|NaN|NA)$/) {
```

Comment:
Could you comment, what this regex does? Also the regex below? 
Do I understand it right, that this one checks the complete string to be either a decimal or NaN or NA?